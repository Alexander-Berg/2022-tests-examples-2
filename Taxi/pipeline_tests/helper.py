import collections
import datetime

from js_pipeline.generated import models

from admin_pipeline.generated.service.swagger.models import api
from admin_pipeline.pipeline import common as pipeline_common
from admin_pipeline.pipeline_tests import views as testing_views


class TestingHelper:
    @staticmethod
    async def test_pipeline(request_body, context):
        assert False, 'test_pipeline is not defined in helper'

    def serialize(self, pipeline: models.Pipeline, add_consumer=False) -> dict:
        assert False, 'test_pipeline is not defined in helper'

    @staticmethod
    def __serialize_tests(tests_results):
        # Keys, starting with '$' is forbidden in mongodb
        def __replace_dollar_rec(obj):
            if isinstance(obj, dict):
                return {
                    key.replace('$', '_$'): __replace_dollar_rec(value)
                    for key, value in obj.items()
                }
            if isinstance(obj, list):
                return [__replace_dollar_rec(value) for value in obj]
            return obj

        return __replace_dollar_rec(
            tests_results.body.tests_results.serialize()['tests'],
        )

    async def enumerate_pipeline_tests(self, consumer, pipeline_name, context):
        tests = [
            api.EnumeratedTest(test['_id'], test['name'], test['scope'])
            async for test in testing_views.TestsView.find_many(
                context.mongo.secondary, {'scope': 'global'}, consumer,
            )
        ]
        togglers = await testing_views.ChoosenTestsHelper.try_find_one(
            context.mongo.secondary,
            {'pipeline_name': pipeline_name},
            consumer,
        )
        if togglers:
            tests.extend(
                [
                    api.EnumeratedTest(
                        test['_id'], test['name'], test['scope'],
                    )
                    async for test in testing_views.TestsView.find_many(
                        context.mongo.secondary,
                        {'name': {'$in': togglers['tests']}},
                        consumer,
                    )
                ],
            )
        return api.TestsEnumerateResponse(tests=tests)

    async def enable_pipeline_tests(self, request, context):
        await testing_views.ChoosenTestsHelper.update_one(
            context.mongo,
            {
                '$and': [
                    {'pipeline_name': request.pipeline_name},
                    {'consumer': request.consumer},
                ],
            },
            {'$set': {'tests': request.body.enabled_tests}},
            upsert=True,
        )

    async def get_pipeline_tests_results(self, pipeline_id, consumer, context):
        # intentionally query master
        result = await testing_views.TestsResultsHelper.try_find_one(
            context.mongo,
            {'pipeline_id': pipeline_id},
            consumer,
            for_model=models.PipelineTestsResults,
        )
        if not result:
            return None
        return models.PipelineTestsResults.deserialize(
            testing_views.drop_id(result),
        )

    async def perform_tests(self, pipeline, consumer, context):
        enumerated_tests = await self.enumerate_pipeline_tests(
            consumer, pipeline.name, context,
        )
        tests = [
            await testing_views.TestsView.get(
                enumerated_test.id, consumer, context,
            )
            for enumerated_test in enumerated_tests.tests
        ]
        if not tests:
            return models.PipelineTestsResults(
                created=datetime.datetime.now(), tests=[],
            )

        enumerated_mocks = await testing_views.MocksView.enumerate_mocks(
            consumer, context,
        )
        resources_mocks = collections.defaultdict(dict)
        prefetched_resources_mocks = collections.defaultdict(dict)
        input_mocks = dict()
        for enumerated_mock in enumerated_mocks.resources_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            resources_mocks[mock.resource][mock.name] = mock.mock
        for enumerated_mock in enumerated_mocks.prefetched_resources_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            prefetched_resources_mocks[mock.resource][mock.name] = mock.mock
        for enumerated_mock in enumerated_mocks.input_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            input_mocks[mock.name] = mock.mock

        enumerated_checks = await testing_views.ChecksView.enumerate_checks(
            consumer, context,
        )
        checks = dict()
        for enumerated_check in enumerated_checks.output_checks:
            check = await testing_views.ChecksView.get(
                enumerated_check.id, consumer, context,
            )
            checks[check.name] = check.check.serialize()

        test_request_body = {
            'pipeline': self.serialize(pipeline, add_consumer=True),
            'tests': [test.serialize() for test in tests],
            'resources_mocks': resources_mocks,
            'prefetched_resources_mocks': prefetched_resources_mocks,
            'input_mocks': input_mocks,
            'output_checks': checks,
        }
        tests_results = await self.test_pipeline(test_request_body, context)
        tzinfo = (
            datetime.datetime.now(datetime.timezone.utc).astimezone().tzinfo
        )
        now = datetime.datetime.now().astimezone(tzinfo).isoformat()
        await testing_views.TestsResultsHelper.update_one(
            context.mongo,
            {'$and': [{'pipeline_id': pipeline.id}, {'consumer': consumer}]},
            {
                '$set': {
                    'pipeline_id': pipeline.id,
                    'consumer': consumer,
                    'created': now,
                    'tests': self.__serialize_tests(tests_results),
                },
            },
            upsert=True,
        )
        return await self.get_pipeline_tests_results(
            pipeline.id, consumer, context,
        )

    async def test_pipeline_request(self, request, context):
        pipeline = await pipeline_common.get_by_id(
            request.body.pipeline_id, self, context,
        )
        return await self.perform_tests(pipeline, request.consumer, context)
