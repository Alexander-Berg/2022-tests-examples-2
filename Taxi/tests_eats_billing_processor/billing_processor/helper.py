import datetime
import json

from testsuite.utils import callinfo


class BillingProcessorTest:
    def __init__(self):
        self._order_nr = '12345'
        self._billing_events = []
        self._columns = ['kind']
        self._expected_transfers = []
        self._should_reschedule = False
        self._expect_fail = False
        self._expected_core_write_back = []

    def for_order_nr(self, order_nr):
        self._order_nr = order_nr
        return self

    def expected_fail(self):
        self._expect_fail = True
        return self

    def insert_billing_event(self, data, input_event_id=1):
        counter = len(self._billing_events) + 1
        self._billing_events.append(
            {
                'input_event_id': input_event_id,
                'order_nr': self._order_nr,
                'external_id': f'auto_{counter}',
                'kind': (
                    'payment'
                    if 'payment' in data.keys()
                    else 'refund'
                    if 'refund' in data.keys()
                    else 'commission'
                ),
                'event_at': datetime.datetime(2021, 3, 26, 17, counter, 0, 0),
                'transaction_date': data['transaction_date'],
                'data': data,
            },
        )
        return self

    def should_reschedule(self):
        self._should_reschedule = True
        return self

    def checking_columns(self, *columns):
        self._columns = columns
        return self

    def expect_transfers(self, *transfers):
        self._expected_transfers.extend(transfers)
        return self

    def expect_core_write_back(self, *events):
        self._expected_core_write_back.extend(events)
        return self

    async def run(self, fixture):
        core_write_back = []

        @fixture.mockserver.json_handler(
            '/eats-billing-storage/billing-storage/create/bulk',
        )
        def _handler(request):
            nonlocal core_write_back
            for item in request.json:
                if item['kind'] == 'CommissionOrder':
                    core_write_back.append(
                        ('commission', item['data']['commission_sum']),
                    )
                else:
                    core_write_back.append(
                        (
                            item['data']['transaction_type'],
                            item['data']['value_amount'],
                            item['data']['commission'],
                        ),
                    )
            response = {'message': 'OK', 'status': 'success'}
            return fixture.mockserver.make_response(json=response, status=200)

        cursor = fixture.pgsql['eats_billing_processor'].cursor()
        for event in self._billing_events:
            data = event['data']
            details = data[event['kind']]
            cursor.execute(
                f"""insert into eats_billing_processor.billing_events
                 (input_event_id, order_nr, external_id, kind, event_at,
                  transaction_date, external_payment_id, client_id,
                  product_type, amount, currency, data)
                 values ({event['input_event_id']}, '{event['order_nr']}',
                         '{event['external_id']}', '{event['kind']}',
                         '{event['event_at']}', '{event['transaction_date']}',
                         '{data['external_payment_id']}',
                         '{data['client']['id']}', '{details['product_type']}',
                          {details['amount']}, '{details['currency']}',
                         '{json.dumps(event['data'])}')""",
            )

        await fixture.stq_runner.eats_billing_processor_billing_processor.call(
            task_id='billing_1',
            kwargs={'order_nr': self._order_nr},
            expect_fail=self._expect_fail,
        )

        try:
            eats_billing_processor_stq = (
                fixture.stq.eats_billing_processor_billing_processor
            )
            call = eats_billing_processor_stq.next_call()
            assert self._should_reschedule
            assert call['id'] == 'billing_1'
        except callinfo.CallQueueEmptyError:
            assert not self._should_reschedule

        cursor.execute(
            f"""select {', '.join(self._columns)}
            from eats_billing_processor.transfers
            where service_order_id='{self._order_nr}'""",
        )
        billing_events = list(cursor)
        assert billing_events == self._expected_transfers

        assert core_write_back == self._expected_core_write_back
