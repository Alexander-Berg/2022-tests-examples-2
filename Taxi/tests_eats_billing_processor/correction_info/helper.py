def column_format(column_name, column_number, title):
    column = {
        'column_name': column_name,
        'column_number': column_number,
        'title': title,
    }
    return column


def make_doc(doc_id, transaction):
    doc = {'doc_id': doc_id, 'data': {'payments': transaction}}
    return doc


class CorrectionInfoTest:
    def __init__(self):
        self._request = None
        self._response = None
        self._order_nr = None
        self._expect_fail = False
        self._expected_status = 200
        self._docs = None

    def request(self, order_nr):
        self._order_nr = order_nr
        self._request = {'order_nr': self._order_nr}
        return self

    def expect_fail(self, status=500):
        self._expect_fail = True
        self._expected_status = status
        return self

    def response(self, amounts, table_format, corrections, docs=None):
        self._response = {
            'order_nr': self._order_nr,
            'amounts': amounts,
            'table_format': table_format,
            'corrections': corrections,
        }
        self._docs = docs
        return self

    async def run(self, fixtures):
        if self._expect_fail:
            fixtures.billing_reports_mock.should_fail()
        else:
            fixtures.billing_reports_mock.docs(self._docs)

        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/correction_info', json=self._request,
        )
        assert response.status == self._expected_status
        if not self._expect_fail:
            amounts = self._response['amounts']
            expect_amounts = response.json()['amounts']
            assert len(amounts) == len(expect_amounts)
            for amount in amounts:
                find = False
                for expect_amount in expect_amounts:
                    if amount == expect_amount:
                        find = True
                        break
                assert find is True

            corrections = self._response['corrections']
            expect_corrections = response.json()['corrections']
            assert len(corrections) == len(expect_corrections)
            for correction in corrections:
                find = False
                for expect_correction in expect_corrections:
                    if correction == expect_correction:
                        find = True
                        break
                assert find is True

            table_format = self._response['table_format']
            expect_table_format = response.json()['table_format']
            assert len(table_format) == len(expect_table_format)
            for column in table_format:
                find = False
                for expect_column in expect_table_format:
                    if column == expect_column:
                        find = True
                        break
                assert find is True
