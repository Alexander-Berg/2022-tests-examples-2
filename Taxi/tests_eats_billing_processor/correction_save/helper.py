COMPLETE_STATUS = 200
INCORRECT_STATUS = 409
ERROR_STATUS = 500

COMPLETE_MESSAGE = 'Корректировка успешно создана'
ERROR_MESSAGE = 'Internal Server Error'


def make_doc(doc_id, transaction):
    doc = {'doc_id': doc_id, 'data': {'payments': transaction}}
    return doc


class CorrectionSaveTest:
    def __init__(self):
        self._request = None
        self._response = None
        self._should_fail = False
        self._docs = None
        self._expect_amount = '0'
        self._expected_status = COMPLETE_STATUS
        self._response = {
            'code': f'{COMPLETE_STATUS}',
            'message': COMPLETE_MESSAGE,
        }

    def request(
            self,
            order_nr,
            login,
            amount,
            currency,
            ticket,
            correction_type,
            correction_group,
            product=None,
            detailed_product=None,
    ):
        self._request = {
            'order_nr': order_nr,
            'login': login,
            'amount': amount,
            'currency': currency,
            'ticket': ticket,
            'correction_group': correction_group,
            'correction_type': correction_type,
        }
        if product is not None:
            self._request['product'] = product
        if detailed_product is not None:
            self._request['detailed_product'] = detailed_product
        return self

    def expected_status(self, status, expected_message):
        self._expected_status = status
        self._response['code'] = f'{status}'
        self._response['message'] = expected_message
        return self

    def should_fail(self, status, code, message):
        self._should_fail = True
        self._expected_status = status
        self._response = {'code': code, 'message': message}
        return self

    def docs(self, docs):
        self._docs = docs
        return self

    def expect_amount(self, amount):
        self._expect_amount = amount
        return self

    async def run(self, fixtures):
        if self._expected_status == ERROR_STATUS:
            fixtures.billing_reports_mock.should_fail()
        else:
            fixtures.billing_reports_mock.docs(self._docs)
        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/correction_save', json=self._request,
        )
        assert response.status == self._expected_status
        assert response.json() == self._response

        if self._should_fail is False:
            cursor = fixtures.pgsql['eats_billing_processor'].cursor()
            cursor.execute(
                f"""select login, ticket, correction_group, correction_type,
                    cast(amount as TEXT) as amount, currency,
                    detailed_product, product
                    from eats_billing_processor.order_accounting_correction""",
            )
            res = cursor.fetchone()
            assert res[0] == self._request['login']
            assert res[1] == self._request['ticket']
            assert res[2] == self._request['correction_group']
            assert res[3] == self._request['correction_type']
            assert res[4] == self._expect_amount
            assert res[5] == self._request['currency']
            if 'detailed_product' in self._request:
                assert res[6] == self._request['detailed_product']
            if 'product' in self._request:
                assert res[7] == self._request['product']

            call_info = (
                fixtures.stq.eats_orders_billing_input_events.next_call()
            )
            order_nr = self._request['order_nr']
            assert call_info['id'] == f'eats_orders_billing_{order_nr}_1'
