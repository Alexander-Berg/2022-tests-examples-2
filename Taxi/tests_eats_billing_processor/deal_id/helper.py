import datetime
import string

import psycopg2.tz

GROCERY_CLIENT_ID = '94062943'

SERVICE_FEE_RUB_CLIENT_ID = '95332016'
SERVICE_FEE_RUB_CONTRACT_ID = '4469918'

SERVICE_FEE_BYN_CLIENT_ID = '97543826'
SERVICE_FEE_BYN_CONTRACT_ID = '5339784'

SERVICE_FEE_KZT_CLIENT_ID = '97543252'
SERVICE_FEE_KZT_CONTRACT_ID = '5339448'

DONATION_CLIENT_ID = '93696285'
DONATION_CONTRACT_ID = '3636412'


def deal_info(client_id, contract_id, mvp):
    info = {'mvp': mvp, 'contract_id': contract_id, 'id': client_id}

    return info


class DealIdTest:
    def __init__(self):
        self._request = None
        self._client_info = {'places': {}, 'couriers': {}, 'pickers': {}}
        self._expected = {'deal_id': None}
        self._expected_status = 200
        self._expected_response = {}
        self._should_fail = False
        self._order_nr = '123456'
        self._expected_data = {'data': {}, 'order_nr': self._order_nr}
        self._token = None

    def on_order_nr(self, order_nr):
        self._order_nr = order_nr
        self._expected_data['order_nr'] = self._order_nr
        return self

    def token(self):
        return self._token

    def on_picker(
            self, picker_id, place_type, delivery_type, assembly_type=None,
    ):
        self._request = {
            'counterparty_id': picker_id,
            'counterparty_type': 'picker',
            'order_nr': self._order_nr,
            'place_type': place_type,
            'delivery_type': delivery_type,
            'timestamp': '2021-03-18T15:00:00+03:00',
        }
        self._expected_data['place_type'] = place_type
        self._expected_data['delivery_type'] = delivery_type

        if assembly_type is not None:
            self._request['assembly_type'] = assembly_type
            self._expected_data['assembly_type'] = assembly_type
        return self

    def on_courier(
            self, courier_id, place_type, delivery_type, assembly_type=None,
    ):
        self._request = {
            'counterparty_id': courier_id,
            'counterparty_type': 'courier',
            'order_nr': self._order_nr,
            'place_type': place_type,
            'delivery_type': delivery_type,
            'timestamp': '2021-03-18T15:00:00+03:00',
        }
        self._expected_data['place_type'] = place_type
        self._expected_data['delivery_type'] = delivery_type

        if assembly_type is not None:
            self._request['assembly_type'] = assembly_type
            self._expected_data['assembly_type'] = assembly_type
        return self

    def on_place(
            self, place_id, place_type, delivery_type, assembly_type=None,
    ):
        self._request = {
            'counterparty_id': place_id,
            'counterparty_type': 'place',
            'order_nr': self._order_nr,
            'place_type': place_type,
            'delivery_type': delivery_type,
            'timestamp': '2021-03-18T15:00:00+03:00',
        }
        self._expected_data['place_type'] = place_type
        self._expected_data['delivery_type'] = delivery_type

        if assembly_type is not None:
            self._request['assembly_type'] = assembly_type
            self._expected_data['assembly_type'] = assembly_type
        return self

    def on_service_fee(
            self,
            place_type,
            delivery_type,
            currency,
            info,
            timestamp=None,
            assembly_type=None,
    ):
        self._request = {
            'place_type': place_type,
            'delivery_type': delivery_type,
            'assembly_type': assembly_type,
            'counterparty_type': 'service_fee',
            'currency': currency,
            'order_nr': self._order_nr,
            'timestamp': timestamp or '2021-03-18T15:00:00+03:00',
        }

        self._expected_data['data']['info'] = info
        self._expected_data['data']['version'] = '1'
        self._expected_data['data']['rule'] = 'default'
        self._expected_data['counterparty_type'] = 'service_fee'
        self._expected_data['counterparty_id'] = info['id']
        self._expected_data['data']['kind'] = 'service_fee'
        self._expected_data['timestamp'] = datetime.datetime(
            2021,
            3,
            18,
            12,
            0,
            0,
            0,
            tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=180, name=None),
        )
        return self

    def on_donation(self, currency, info, timestamp=None):
        self._request = {
            'counterparty_type': 'donation',
            'order_nr': self._order_nr,
            'currency': currency,
            'timestamp': timestamp or '2021-03-18T15:00:00+03:00',
        }
        self._expected_data['data']['info'] = info
        self._expected_data['data']['version'] = '1'
        self._expected_data['data']['rule'] = 'donation'
        self._expected_data['counterparty_type'] = 'donation'
        self._expected_data['counterparty_id'] = info['id']
        self._expected_data['data']['kind'] = 'donation'
        self._expected_data['timestamp'] = datetime.datetime(
            2021,
            3,
            18,
            12,
            0,
            0,
            0,
            tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=180, name=None),
        )
        return self

    def using_client_info(
            self,
            client_info,
            place_id=None,
            courier_id=None,
            picker_id=None,
            rule=None,
    ):
        counter = sum(
            map(
                lambda x: 0 if x is None else 1,
                [place_id, courier_id, picker_id],
            ),
        )
        if counter == 0:
            raise ValueError('place_id, courier_id or picker_id expected')
        elif counter != 1:
            raise ValueError(
                'Only one of place_id, courier_id and picker_id allowed',
            )

        counterparty_id = None
        client_dict = None
        if place_id is not None:
            counterparty_id = place_id
            client_dict = self._client_info['places']
            self._expected_data['data']['kind'] = 'place'
            self._expected_data['counterparty_type'] = 'place'
        elif courier_id is not None:
            counterparty_id = courier_id
            client_dict = self._client_info['couriers']
            self._expected_data['data']['kind'] = 'courier'
            self._expected_data['counterparty_type'] = 'courier'
        else:
            counterparty_id = picker_id
            client_dict = self._client_info['pickers']
            self._expected_data['data']['kind'] = 'picker'
            self._expected_data['counterparty_type'] = 'picker'

        self._expected_data['data']['info'] = client_info
        self._expected_data['data']['version'] = '1'
        self._expected_data['data']['rule'] = '' if rule is None else rule
        self._expected_data['counterparty_id'] = counterparty_id
        self._expected_data['timestamp'] = datetime.datetime(
            2021,
            3,
            18,
            12,
            0,
            0,
            0,
            tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=180, name=None),
        )

        client_dict[counterparty_id] = client_info

        return self

    def expected_rule_name(self, rule_name):
        self._expected_data['data']['rule'] = rule_name
        return self

    def should_fail(self, status, code, message):
        self._should_fail = True
        self._expected_status = status
        self._expected_response = {'code': code, 'message': message}
        return self

    async def run(self, fixtures):
        fixtures.client_info_mock(self._client_info)

        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/deal_id', json=self._request,
        )

        if self._should_fail:
            assert response.status == self._expected_status
            assert response.json() == self._expected_response
        else:
            cursor = fixtures.pgsql['eats_billing_processor'].cursor()
            cursor.execute(
                f"""select token, data, counterparty_type, counterparty_id,
                    order_nr, timestamp, rule_name
                    from eats_billing_processor.deals""",
            )
            res = cursor.fetchone()
            # if res:
            self._token = res[0]
            assert res[1] == self._expected_data['data']
            assert res[2] == self._expected_data['counterparty_type']
            assert res[3] == self._expected_data['counterparty_id']
            assert res[4] == self._expected_data['order_nr']
            assert res[5] == self._expected_data['timestamp']
            assert all(
                c in string.hexdigits for c in response.json()['deal_id']
            )
