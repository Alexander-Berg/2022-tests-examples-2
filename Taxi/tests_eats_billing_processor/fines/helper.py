class OrderFinesTest:
    def __init__(self):
        self._request = None
        self._expected = {'fines': []}
        self._expected_status = 200
        self._expected_response = {'code': '200', 'message': 'OK'}
        self._create_expected_response = {'event_id': '1'}
        self._expected_create_request = None
        self._expected_appeals = []

    def on_request(self, order_nr):
        self._request = {'order_nr': order_nr}
        return self

    def on_request_by_id(self, fine_id):
        self._request = {'id': fine_id}
        return self

    def appeal_request(self, fine_id, ticket):
        self._request = {'fine_id': fine_id, 'ticket': ticket}
        return self

    def expect_order_fine(
            self,
            fine_id,
            amount,
            currency,
            is_appealed,
            ticket=None,
            fine_reason_id=None,
            fine_reason=None,
    ):
        order_fine = {
            'fine_id': fine_id,
            'amount': amount,
            'currency': currency,
            'is_appealed': is_appealed,
        }
        if ticket is not None:
            order_fine['ticket'] = ticket
        if fine_reason_id is not None:
            order_fine['fine_reason_id'] = fine_reason_id
        if fine_reason is not None:
            order_fine['fine_reason'] = fine_reason
        self._expected['fines'].append(order_fine)
        return self

    def expect_fine_by_id(
            self,
            fine_id,
            order_nr,
            amount,
            currency,
            is_appealed,
            fine_reason_id,
            fine_reason,
            ticket=None,
    ):
        fine_by_id = {
            'fine_id': fine_id,
            'order_nr': order_nr,
            'amount': amount,
            'currency': currency,
            'is_appealed': is_appealed,
            'fine_reason_id': fine_reason_id,
            'fine_reason': fine_reason,
        }
        if ticket is not None:
            fine_by_id['ticket'] = ticket
        self._expected = {'fine': fine_by_id}
        return self

    def expect_fine_appeal_event(self, **kwargs):
        self._expected_create_request = kwargs
        return self

    def should_fail(self, status, code, message=None):
        self._expected_status = status
        self._expected_response = {'code': code}
        if message is not None:
            self._expected_response['message'] = message
        return self

    def expect_appeal(self, fine_id, ticket, amount):
        self._expected_appeals.append((fine_id, ticket, amount))
        return self

    async def run(self, fixtures):
        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/order_fines', json=self._request,
        )

        assert response.status == 200
        assert response.json() == self._expected

    async def run_by_id(self, fixtures):
        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/order_fines/by_id', json=self._request,
        )

        assert response.status == self._expected_status
        if response.status == 200:
            assert response.json() == self._expected
        else:
            assert response.json()['code'] == self._expected_response['code']
            if 'message' in self._expected_response:
                assert (
                    response.json()['message']
                    == self._expected_response['message']
                )

    async def appeal_run(self, fixtures, pgsql):
        @fixtures.mockserver.json_handler('/eats-billing-processor/v1/create')
        def _handler(req):
            assert req.json['data'] == self._expected_create_request
            return fixtures.mockserver.make_response(
                status=200,
                json=self._create_expected_response,
                headers={},
                content_type='application/json',
            )

        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/order_fines/appeal', json=self._request,
        )
        assert response.status == self._expected_status
        assert response.json() == self._expected_response

        cursor = pgsql['eats_billing_processor'].cursor()
        cursor.execute(
            f"""select cast(fine_id as text) as fine_id,
            ticket as ticket,
            cast(amount as text) as amount
            from eats_billing_processor.appeals""",
        )
        result_appeals = list(cursor)
        assert result_appeals == self._expected_appeals
