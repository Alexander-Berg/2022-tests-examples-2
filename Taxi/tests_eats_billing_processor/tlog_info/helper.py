def column_format(column_name, column_number, title):
    column = {
        'column_name': column_name,
        'column_number': column_number,
        'title': title,
    }
    return column


def make_doc(doc_id, transaction):
    doc = {'doc_id': doc_id, 'data': {'payments': transaction}}
    return doc


class TLogInfoTest:
    def __init__(self):
        self._request = None
        self._response = None
        self._order_nr = None
        self._expected_status = 200
        self._times_called = 0
        self._async_fail = False
        self._docs = None

    def request(self, order_nr):
        self._order_nr = order_nr
        self._request = {'order_nr': self._order_nr}
        return self

    def async_fail(self):
        self._async_fail = True
        self._times_called = 3
        self._expected_status = 500
        return self

    def response(self, table_format, docs):
        self._docs = docs
        trs = []
        for doc in docs:
            trs.append(doc['data']['payments'])
        self._response = {
            'order_nr': self._order_nr,
            'table_format': table_format,
            'transactions': trs,
        }
        self._times_called = 1
        return self

    async def run(self, fixtures):
        if self._async_fail:
            fixtures.billing_reports_mock.should_fail()
        else:
            fixtures.billing_reports_mock.docs(self._docs)

        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/tlog_info', json=self._request,
        )

        assert fixtures.billing_reports_mock.times_called == self._times_called
        assert response.status == self._expected_status

        if self._async_fail is False:
            table_format = self._response['table_format']
            expect_table_format = response.json()['table_format']

            for column in table_format:
                find = False
                for expect_column in expect_table_format:
                    if column == expect_column:
                        find = True
                        break
                assert find is True

            assert (
                response.json()['transactions']
                == self._response['transactions']
            )
