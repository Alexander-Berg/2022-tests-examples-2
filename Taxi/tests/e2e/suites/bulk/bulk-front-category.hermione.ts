import {openChangeFrontCategoryBulkPage} from 'tests/e2e/helper/bulk';
import {parseSpreadSheetJSON} from 'tests/e2e/helper/parse-spreadsheet';
import {describe, expect, it} from 'tests/hermione.globals';

describe('Балковые действия для фронт-категорий', function () {
    it('Добавить ФК во все товары', async function () {
        await openChangeFrontCategoryBulkPage(this, 'all', 'add');
        await this.browser.clickInto('fc_row_front_category_code_1_0', {waitRender: true});
        await this.browser.clickInto(['fc_row_front_category_code_5_0', 'checkbox'], {waitRender: true});
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('bulk-success-alert');
        await this.browser.clickInto('import-info-update', {waitRender: true});
        await this.browser.assertImage('import-info-update');
    });

    it('Удалить ФК у всех товаров', async function () {
        await openChangeFrontCategoryBulkPage(this, 'all', 'remove');
        await this.browser.clickInto('fc_row_front_category_code_1_0', {waitRender: true});
        await this.browser.clickInto(['fc_row_front_category_code_5_0', 'checkbox'], {waitRender: true});
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('bulk-success-alert');
        await this.browser.clickInto('import-info-update', {waitRender: true});
        await this.browser.assertImage('import-info-update');
    });

    it('Добавить нескольким товарам фронт-категорию', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001, 10000002], 'add');
        await this.browser.clickInto('fc_row_front_category_code_1_1', {waitRender: true});
        await this.browser.clickInto(['fc_row_front_category_code_6_1', 'checkbox'], {waitRender: true});
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('bulk-success-alert');
        await this.browser.clickInto('download-xlsx-link', {waitRender: true});
        const file = await this.browser.getDownloadedFile('exported-products.xlsx', {purge: true});
        expect(parseSpreadSheetJSON(file)).to.matchSnapshot(this);
    });

    it('Убрать у одного товара фронт-категорию', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001], 'remove');
        await this.browser.clickInto('fc_row_front_category_code_1_0', {waitRender: true});
        await this.browser.clickInto(['fc_row_front_category_code_5_1', 'checkbox'], {waitRender: true});
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button');
        await this.browser.waitForTestIdSelectorInDom('bulk-success-alert');
        await this.browser.clickInto('download-xlsx-link', {waitRender: true});
        const file = await this.browser.getDownloadedFile('exported-products.xlsx', {purge: true});
        expect(parseSpreadSheetJSON(file)).to.matchSnapshot(this);
    });

    it('Показ неактивных категорий при добавлении во фронт-категории', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001], 'add');

        await this.browser.performScroll(['tree-table', 'list-holder']);
        await this.browser.assertImage('tree-table');

        await this.browser.clickInto(['action-bar', 'show-inactive'], {waitRender: true});

        await this.browser.performScroll(['tree-table', 'list-holder']);
        await this.browser.assertImage('tree-table');
    });

    it('Показ неактивных категорий при удалении из фронт-категорий', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000004], 'remove');

        await this.browser.performScroll(['tree-table', 'list-holder']);
        await this.browser.assertImage('tree-table');

        await this.browser.clickInto(['action-bar', 'show-inactive'], {waitRender: true});

        await this.browser.performScroll(['tree-table', 'list-holder']);
        await this.browser.assertImage('tree-table');
    });

    it('Поиск активной фронт-категории по названию при добавлении', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001], 'add');

        const frontCategoryName = 'Как жизни презирает';

        await this.browser.typeInto('search', frontCategoryName);
        await this.browser.waitUntilRendered();

        await this.browser.assertImage(['list-holder', 'fc_row_front_category_code_1_0']);
    });

    it('Поиск неактивной фронт-категории по названию при удалении', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000005], 'remove');

        const frontCategoryName = 'Не предпочел';

        await this.browser.clickInto('show-inactive');
        await this.browser.typeInto('search', frontCategoryName);
        await this.browser.waitUntilRendered();

        await this.browser.assertImage(['list-holder', 'fc_row_front_category_code_7_4']);
    });

    it('Страница массового добавления фронт-категории', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001, 10000002], 'add');

        await this.browser.assertImage('base-layout');
    });

    it('Страница массового удаления фронт-категории', async function () {
        await openChangeFrontCategoryBulkPage(this, [10000001, 10000002], 'remove');

        await this.browser.assertImage('base-layout');
    });
});
