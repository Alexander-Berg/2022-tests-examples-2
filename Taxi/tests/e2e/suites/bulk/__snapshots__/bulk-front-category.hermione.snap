exports[`Балковые действия для фронт-категорий Добавить нескольким товарам фронт-категорию 1`] = [
  {
    "id": 10000001,
    "front_category": "front_category_code_5_0; front_category_code_5_1; front_category_code_5_2; front_category_code_6_1"
  },
  {
    "id": 10000002,
    "front_category": "front_category_code_5_3; front_category_code_5_4; front_category_code_6_0; front_category_code_6_1"
  }
];
exports[`Балковые действия для фронт-категорий Убрать у одного товара фронт-категорию 1`] = [
  {
    "id": 10000001,
    "front_category": "front_category_code_5_0; front_category_code_5_2"
  }
];
